frame/fee-market/src/lib.rs

```diff
@@ -215,20 +215,24 @@ pub mod pallet {
 				},
 				WithdrawReasons::all(),
 			);
-
-			<RelayersMap<T>>::insert(&who, Relayer::new(who.clone(), lock_collateral, fee));
-			<Relayers<T>>::append(who.clone());
+			// Store enrollment detail information.
+			<RelayersMap<T>>::insert(
+				&who,
+				Relayer::new(who.clone(), lock_collateral, fee, order_capacity),
```

Comment:
`order_capacity` looks like a cache of `(lock_collateral / T::CollateralPerOrder::get()).saturated_into::<u32>()`, will it be better if we transform it to a pure internal function? The benefit is that the cached does not need be updated, and it will not be affected by the changes of `lock_collateral` or ` T::CollateralPerOrder::get()`.

Reply:
Yes, you can see it as a sign to the locked collateral. If we use a purely internal function, it may look like:
```rs
fn cal_order_capacity(who: & AccountId) -> u32 {
      Pallet::relayer(who).collateral / T::CollateralPerOrder::get()
}
```
Then, we have to call this function everywhere we. So I prefer the current implementation.

We only need to update this cache only in two cases, not so much in my view: 
1. Update locked collateral, a little complicated.
2. Slash collateral, slash CollateralPerOrder per message, order_capacity -1, this is easy to maintain.

Reply:
> There are two hard things in computer science: cache invalidation, naming things, and off-by-one errors.

Don't underestimate the extra difficulty bring by cache management especially considering the audit burden and security requirement.

For the performance and ease of use, I can not tell difference between. Self.cal_order_capacity(account_id) and <RelayersMap<T>>::get(account_id).order_capacity

Reply:
Another thing is that `order_capacity` is useful to show the ability to relay when the relayer accepts an order or confirms an order. It's much harder to achieve this if we use the pure function. IMO.

Reply:
We can use rpc if the purpose is for ease of use for apps or bridger.

Reply:
No, the apps or bridger can know the order_capacity thought RPC `get_relayer`. `The useful` means +1 when accepts an order in the callback, and -1 when confirms an order in the back. Then update_market select proper assgined relayers according to `order_capacity`.

See: 
1. https://github.com/darwinia-network/darwinia-common/blob/update-fee-market/frame/fee-market/src/s2s/callbacks.rs#L37
2. https://github.com/darwinia-network/darwinia-common/blob/update-fee-market/frame/fee-market/src/s2s/payment.rs#L173